client/src/common/components/Dropdown.jsx

```diff
@@ -65,7 +79,7 @@ class Dropdown extends PureComponent {
       <div className="dropdown">
         <button
           className={classNames(buttonClassName)}
-          onClick={this.handleDropdownVisibility}
+          onClick={isVisible ? this.hideDropdown : this.showDropdown}
```

Comment:
Consider extracting this before the `return (` as `const toggle = isVisible ? this.hide : this.show`.